### PR TITLE
Implement get_by_ids

### PR DESCRIPTION
**Pull request by beccadaniel**
## Why make this change?
This change introduces a new function `get_by_ids`.

## What is this change?
In this PR, we implement `get_by_ids`, a function that gets documents by the IDs provided.

## Test
image

### CODE REVIEW

**Review by yorek — APPROVED** [review]

File: libs/community/langchain_community/vectorstores/sqlserver.py [hunk 1]
```diff
@@ -331,6 +332,45 @@ def from_texts(
...
+        documents = []
+
+        if ids is None or len(ids) == 0:
+            logging.info(EMPTY_IDS_ERROR_MESSAGE)
+        else:
+            result = self._get_documents_by_ids(ids)
+            for item in result:
+                documents.append(
+                    Document(
+                        id=item.custom_id,
```
Comment (Aniruddh25) [inline libs/community/langchain_community/vectorstores/sqlserver.py]: how can we ensure item has a custom_id, content or metadata if its not typed?

File: libs/community/tests/unit_tests/vectorstores/test_sqlserver.py [hunk 2]
```diff
@@ -349,6 +349,23 @@ def test_that_multiple_vector_stores_can_be_created(
...
+    """Test that `get_by_ids` returns documents."""
+    ids = store.add_texts(texts, metadatas)
+
+    # Append non-existent id to list of IDs to get.
+    ids.append("200")
+    documents = store.get_by_ids(ids)
+
+    # Assert that the length of documents returned
+    # is the same as length of inserted texts.
+    assert len(documents) == len(metadatas)
```
Comment (Aniruddh25) [inline libs/community/tests/unit_tests/vectorstores/test_sqlserver.py:433]: can also assert len(documents) != len(ids) since doc for 200 shouldnt be there

**Review by Aniruddh25 — APPROVED** [review]
LGTM

File: libs/community/langchain_community/vectorstores/sqlserver.py (hunk 1 above)
Comment (beccadaniel) [reply libs/community/langchain_community/vectorstores/sqlserver.py]: `item` is a result from the DB where we've selected those specific columns: `custom_id`, `content`, `metadata`.
If `item` itself is not None, then we will be getting those columns. Hence why we are able to use though it's not typed.

For validation on if e.g `item.custom_id` results to `None`, the `Document` class does that validation.

I'll update the code to check that `item` is not `None`.

**Review by Aniruddh25 — APPROVED** [review]

**Pull request merged**